lightningd/gossip_control.c

```diff
@@ -341,15 +351,42 @@ static void json_getroute(struct command *cmd, const char *buffer, const jsmntok
 		return;
 	}
 
-	u8 *req = towire_gossip_getroute_request(cmd, &source, &destination, msatoshi, riskfactor*1000, cltv);
+	if (fuzztok &&
+	    !json_tok_double(buffer, fuzztok, &fuzz)) {
+		command_fail(cmd, "'%.*s' is not a valid double",
+			     (int)(fuzztok->end - fuzztok->start),
+			     buffer + fuzztok->start);
+		return;
+	}
+	if (!(0.0 <= fuzz && fuzz <= 100.0)) {
+		command_fail(cmd,
+			     "fuzz must be in range 0.0 <= %f <= 100.0",
+			     fuzz);
+		return;
+	}
+	/* Convert from percentage */
+	fuzz = fuzz / 100.0;
+	fuzz_scaled = (u32) (fuzz * 4294967296.0);
```

Comment:
This still means 100% == 0!

Let's fix this properly and do towire_double/fromwire_double.  It can just memcpy, since our inter-daemon comms is all same-architecture, and I can't find a portable way to encode a double other than printf/scanf :(